base/float16.jl

```diff
@@ -1,11 +1,11 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 function convert(::Type{Float32}, val::Float16)
-    ival::UInt32 = reinterpret(UInt16, val)
```

Comment:
does this also deprecate this form, or did you just add `local` because you're deprecating the no-assignment form used for `ret` ?